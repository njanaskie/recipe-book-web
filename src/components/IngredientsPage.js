import React from 'react'
import AddIngredientForm from './AddIngredientForm'
import IngredientsList from './IngredientsList'

const IngredientsPage = () => {

    return (
        <div>
            <AddIngredientForm />
            <IngredientsList />
        </div>
    )

};

export default IngredientsPage;