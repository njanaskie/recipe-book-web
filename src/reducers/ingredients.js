
// Ingredients Reducer

const ingredientsReducerDefaultState = [];

const ingredientsReducer = (state = ingredientsReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_INGREDIENT':
            return [
                ...state,
                action.ingredient
            ];
        case 'REMOVE_INGREDIENT':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_INGREDIENT':
            return state.map((ingredient) => {
                if (ingredient.id === action.id) {
                    return {
                        ...ingredient,
                        ...action.updates
                    };
                } else {
                    return ingredient;
                };
            });
        case 'SET_INGREDIENTS':
            return action.ingredients;
        default:
            return state;
    }
}

export default ingredientsReducer;