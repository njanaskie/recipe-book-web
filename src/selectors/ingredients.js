
export default (ingredients) => {
    return ingredients
    .sort((a, b) => {
        if (a.category === b.category) {
            return b.category - a.category
        }
         return a.category > b.category ? 1 : -1
    })
}