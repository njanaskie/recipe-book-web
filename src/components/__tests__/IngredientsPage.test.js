import React from 'react'
import { shallow, mount } from 'enzyme'
import IngredientsPage from '../IngredientsPage'
import { ReactTinyLink } from 'react-tiny-link'
import ingredients from '../../../tests/fixtures/ingredients'
import IngredientsContext from '../../../context/ingredients-context'

test('should render IngredientsPage correctly',() => {
    const wrapper = shallow(<IngredientsPage />)
    expect(wrapper).toMatchSnapshot()
});