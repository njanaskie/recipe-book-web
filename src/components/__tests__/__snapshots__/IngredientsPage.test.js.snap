// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render IngredientsPage correctly 1`] = `
<ContextProvider
  value={
    Object {
      "pantryDispatch": [Function],
      "pantryIngredients": Array [],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "ingredients": Array [],
      }
    }
  >
    <h3>
      Ingredients Page
    </h3>
    <AddIngredientForm />
    <IngredientsList />
  </ContextProvider>
</ContextProvider>
`;
